From tim.gardner@canonical.com  Fri May  1 14:02:00 2009
From: tim.gardner@canonical.com
Date: Wed, 29 Apr 2009 15:01:43 -0600
Subject: Staging: agnx: mac80211 - unify config_interface and bss_info_changed
To: greg@kroah.com
Cc: Tim Gardner <tim.gardner@canonical.com>
Message-ID: <1241038904-24492-3-git-send-email-tim.gardner@canonical.com>


From: Tim Gardner <tim.gardner@canonical.com>

The commit 'mac80211: unify config_interface and bss_info_changed' from
Johannes Berg <johannes@sipsolutions.net> removed the config_interface structure
tag from struct ieee80211_ops. The BSSID detection functionality migrated to
ieee80211_ops.bss_info_changed.

Renamed agnx_config_interface() to agnc_bss_info_changed() with suitable
refactoring.

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/agnx/pci.c |   17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

--- a/drivers/staging/agnx/pci.c
+++ b/drivers/staging/agnx/pci.c
@@ -302,9 +302,10 @@ static int agnx_config(struct ieee80211_
 	return 0;
 }
 
-static int agnx_config_interface(struct ieee80211_hw *dev,
+static void agnx_bss_info_changed(struct ieee80211_hw *dev,
 				 struct ieee80211_vif *vif,
-				 struct ieee80211_if_conf *conf)
+				 struct ieee80211_bss_conf *info,
+				 u32 changes)
 {
 	struct agnx_priv *priv = dev->priv;
 	void __iomem *ctl = priv->ctl;
@@ -312,17 +313,17 @@ static int agnx_config_interface(struct 
 
 	spin_lock(&priv->lock);
 
-	if (memcmp(conf->bssid, priv->bssid, ETH_ALEN)) {
-		agnx_set_bssid(priv, conf->bssid);
-		memcpy(priv->bssid, conf->bssid, ETH_ALEN);
-		hash_write(priv, conf->bssid, BSSID_STAID);
+	if ((changes & BSS_CHANGED_BSSID) ||
+		memcmp(info->bssid, priv->bssid, ETH_ALEN)) {
+		agnx_set_bssid(priv, info->bssid);
+		memcpy(priv->bssid, info->bssid, ETH_ALEN);
+		hash_write(priv, info->bssid, BSSID_STAID);
 		sta_init(priv, BSSID_STAID);
 		/* FIXME needed? */
 		sta_power_init(priv, BSSID_STAID);
 		agnx_write32(ctl, AGNX_BM_MTSM, 0xff & ~0x1);
 	}
 	spin_unlock(&priv->lock);
-	return 0;
 } /* agnx_config_interface */
 
 
@@ -421,7 +422,7 @@ static struct ieee80211_ops agnx_ops = {
 	.add_interface		= agnx_add_interface,
 	.remove_interface	= agnx_remove_interface,
 	.config			= agnx_config,
-	.config_interface	= agnx_config_interface,
+	.bss_info_changed	= agnx_bss_info_changed,
 	.configure_filter	= agnx_configure_filter,
 	.get_stats		= agnx_get_stats,
 	.get_tx_stats		= agnx_get_tx_stats,
